src/init_gui.py

```diff
@@ -0,0 +1,80 @@
+import lvgl as lv
+import sys
+import os
+
+def init_gui():

```

Comment:
@uraich There is a different script [`display_driver.py`](https://github.com/lvgl/lv_binding_micropython/blob/master/lib/display_driver.py) under lib which is doing a very similar thing - trying to guess the architecture and initialized a "default" display driver. It works today with the js port, with the unix port and with ESP32. In the future we might add STM32 too.

Would you consider integrating ttgo display into `display_driver.py` and using it instead of `init_gui.py`?

Reply:
In fact I modified the init_gui script and it is even closer to display_driver now. I have added a few methods:

- method to set screen width and hight (works only on the SDL driver)
- method to read the type of driver used.
- method to set the screen orientation for the ili9341 driver
However I also split the script into two, putting the last 2 lines in display_driver into a new script. This allows to use a default initialization but also to fine-tune if needed. 
I will try to finish this and upload it for you to have a look

Reply:
Before creating a pull request I uploaded my display_driver.py to github for you to have a look:
https://github.com/uraich/lv_mpy_examples/tree/main/lib
Normally you would import display_driver.py which provides default initialization parameters.
You can however also import GUI.py and tweak the width, height and screen orientation parameters before initializing
Please tell me what you think of it.

Reply:
I understand the motivation to split this into two files, giving the user the option to set the driver parameters.  

- Currently your `driver` constructor receives no arguments. 
Why did you choose to use setters instead of receiving named arguments in the constructor?
Instead of setters your could receive named arguments in the constructor and you would not even need an "init_gui" function, since the constructor could do all the work (or just call "init_gui" as the last step).
Or is there some reason to first create an "inactive" driver, and initialize it only later?

- I'm not sure about the name "GUI.py" for that module. It's not a GUI, it's the driver initialization.  
How about something line "display_driver_utils.py" instead of "GUI"? so you could: `import driver from display_driver_utils` inside display_driver.py.

Reply:
Everything accepted. If I continue to work with you I may even learn Python on my old days!
Please have a look at the corrected version: https://github.com/uraich/lv_mpy_examples/tree/main/lib
Btw: When replacing the SDL driver initialization in demo_widgets.py by this new initialization, then the program will run with screen size 480x320 pixel resolution in SDL and with 320x240 on the ili9341. This of course also requires touch panel calibration in landscape mode.
I everything is ok like this, then I would create a new pull request.

Reply:
A few comments about the corrected version:

- `orientation="LANDSCAPE"` is not so clean. What if someone mistakenly mistype it or type it in lower case?  
A slightly cleaner approach would be to define constants for that purpose, such that the user would get a syntax error in case of mistyping it.
It would also be more clear what valued are expected in "orientation" arguments.
You can see an example [on ili9XXX driver](https://github.com/lvgl/lv_binding_micropython/blob/9e45d083d58c956d351d770339e2edda984f9447/driver/esp32/ili9XXX.py#L76) where "rot" is a named argument but is expected to receive either `PORTRAIT` or `LANDSCAPE` which are integer constants.  
Just define these constants before the class and make sure the argument matches one of them.
Another option, since there are only two options (Portrait/Landscape), is just to pass a boolean argument.

- Bug: in case of ili9341 in Landscape mode, `self.type` is not being set. It is only set on ili9341 Portrait mode.

- Why do you provide all the arguments (miso, mosi, clk etc.) to ili9341 and xpt2046 instead of leaving the default ones? (unless there is a special reason to pass them, for setting orientation for example)

- If you want to support different orientations for the touch panel, you can provide different calibration arguments for xpt2046 in case of landscape mode. Currently the default "cal_X" arguments in xpt2046 match the portrait mode of a common ili9341 display.

- In "init_gui" you call "init_gui_esp32()", "init_gui_esp32()" but your try to initialize ttgo directly. It would be nicer to create an "init_ttgo" function and call it.

- Since a ttgo is also an ESP32, wouldn't it run both init_gui_esp32() and init_ttgo()? Perhaps it's better to prioritize ttgo initialization, and once an initialization is done, not try the next one. You can just put ttgo first, and add a "return" statement  after each "init_gui_XXX" call such that if one is successful, the init_gui would return instead of trying the next display driver.

Reply:
- replaced string by boolean as proposed
- bug corrected
- I need to modify the parameters in order to be able to test. My hardware has fixed connections which are reflected by these changes:
https://docs.wemos.cc/en/latest/d1_mini_shield/tft_2_4.html
The other option would be to modify the default parameters on the driver, which I wanted to avoid. 
For the pull request I will change this back to defaults parameters (see comments) and put my parameters in comments instead
- initialization for landscape and portrait mode already have different calibration parameters
- t-watch initialization is now a function as all the others
- t-watch initialization done first as proposed and ili9341 initialization is skipped if t-watch init was successful.
I modified the name init_esp32 to init_ili934 to distinguish both eps32 initializations. The t-watch uses an ST7789V display driver and an FT6236U touch controller

Reply:
Looks good! I haven't tested it yet, but I think it's ready for PR.
For keeping backward compatibility with existing display_driver.py, please change the default width/height to 480x320, as is the [default for SDL today](https://github.com/lvgl/lv_binding_micropython/blob/9e45d083d58c956d351d770339e2edda984f9447/lib/display_driver.py#L19-L20).